euphoria-spark/src/main/java/cz/seznam/euphoria/spark/SparkSerializerFactory.java

```diff
@@ -0,0 +1,155 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.seznam.euphoria.spark;
+
+import cz.seznam.euphoria.core.executor.storage.SerializerFactory;
+import org.apache.spark.serializer.DeserializationStream;
+import org.apache.spark.serializer.SerializationStream;
+import org.apache.spark.serializer.SerializerInstance;
+import scala.reflect.ClassTag;
+import scala.reflect.ClassTag$;
+
+import java.io.EOFException;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+class SparkSerializerFactory implements SerializerFactory {
+
+  // ~ adapts spark's SerializerInstance to
+  // euphoria's SerializerFactory.Serializer
+  static class SparkSerializer implements Serializer {
+
+    // ~ adapts spark's SerializationStream to
+    // euphoria's SerializerFactory.Serializer.OutputStream
+    static class SparkSerializerOutputStream implements OutputStream {
+      private final SerializationStream sparkSerializationStream;
+      private ClassTag type;
+
+      SparkSerializerOutputStream(SerializationStream s) {
+        sparkSerializationStream = Objects.requireNonNull(s);
+      }
+
+      @SuppressWarnings("unchecked")
+      @Override
+      public void writeObject(Object o) {
+        if (type == null) {
+          type = ClassTag$.MODULE$.apply(o.getClass());
+        }
+        sparkSerializationStream.writeObject(o, type);
+      }
+
+      @Override
+      public void flush() {
+        sparkSerializationStream.flush();
+      }
+
+      @Override
+      public void close() {
+        sparkSerializationStream.close();
+      }
+    }
+
+    // ~ adopts spark's DeserializationStream to
+    // euphoria's SerializerFactory.Serializer.InputStream
+    static class SparkSerializerInputStream implements InputStream {
+      private final DeserializationStream sparkDeserializationStream;
+
+      private boolean streamFinished;
+      private Object next;
+
+      SparkSerializerInputStream(DeserializationStream s) {
+        sparkDeserializationStream = s;
+      }
+
+      @Override
+      public Object readObject() {
+        while (true) {
+          if (next != null) {
+            Object n = next;
+            next = null;
+            return n;
+          }
+          if (streamFinished) {
+            throw new NoSuchElementException("End of file reached!");
+          }
+          tryReadNext();
+        }
+      }
+
+      @SuppressWarnings("BC_IMPOSSIBLE_INSTANCEOF")
```

Comment:
I think this was renamed to `SuppressFBWarnings` to avoid collision with the `java.lang` version. Can we also add a little note about why we suppress the warning?